docs/modeling-your-data/modeling-your-data-with-dbt/dbt-event-splitting-model/index.md

```diff
@@ -0,0 +1,111 @@
+  - `models_folder` _(optional - string)_: Folder under `models/` to place the models, default `snowplow_split_events`
+- `events` _(required - array)_:
+  - `event_1` _(required - object)_:
+    - `event_name` _(required - string)_: Name of the event type, value of the event_name column in your warehouse
+    - `event_columns` _(optional (>=1 of) - array)_: Array of strings of flat column names from the events table to include to include in the model
+    - `self_describing_event_schema` _(optional (>=1 of) - string)_: `iglu:com.` type url for the self-describing event to include in the model
+    - `context_schemas` _(optional (>=1 of) - array)_: Array of strings of `iglu:com.` type url(s) for the context/entities to include in the model
+    - `context_aliases` _(optional - array)_: Array of strings of prefix to the column alias for context/entities
+    - `table_name` _(optional - string)_: Name of the model, default is the event_name and major version number
+  - `event_2` _(option - object)_
+  - ...
+  - `event_n` _(option - object)_
+- `users` _(option - array)_: Array of strings of schemas for your user contexts to add to your users table as columns, if not provided will not generate users model
+
+An example configuration can be found in the `utils/example_event_split_config.json` file.
+
+### Producing your models
+To produce your models you need to run `python utils/snowplow_split_events_model_gen.py path/to/your/config` from the root of your dbt project. This will produce one `.sql` file for each of the event names specified in the `events` part of your configuration, one `.sql` file for the combined filtered events table if a name was provided, and one `.sql` file for your users table if schema(s) were provided. These files will be in your `models` folder in the sub-folder specified in your config. 
+
+:::info
+Custom error messages have been added to the script to try and catch any issues and provide suggested resolutions to any issues such as invalid configurations or failing validation of schemas. If you persist in getting errors when validating schemas you believe you be correct, you can disable this validation by setting `validate_schemas` to `false` in your config. 
+:::
+
+### Adding new models
+To add new models for new types of events, simply add them to your config and run the model again. Note that `overwrite` must be set to `true` for your filtered events table model to include these new events, and that this will overwrite any manual changes you made to the other models generated by this script.
+
+### Updating your models
+If you wish to update your models, such as adding a new context or removing a column, you must update your config file and run the script again, ensuring the `overwrite` value is set to `true`. Note that this will remove any custom changes you made to the model file itself and you will need to re-add these.
```

Comment:
Should this value ever not be `true`? Just thinking this through, what happens when we don't overwrite? Does it just get skipped?

Reply:
That specific model file gets skipped yes, I can see a case when you might add a new event to your config but you have made some custom changes to the ones you made last time (maybe there's a few columns they didn't need to keep from the context). In that case you don't want to overwrite exisitng models with the same name. It just checks if the file exists and if it does it doesn't write it. It's true by default anyway so it's only if they have specified `false` in the config?